compendium/LogServiceImpl/src/LogServiceImpl.cpp

```diff
@@ -101,15 +195,78 @@ namespace cppmicroservices
                             std::string const& message,
                             const std::exception_ptr ex)
         {
-            std::string full_message = message;
-            full_message = message + GetServiceReferenceInfo(sr) + GetExceptionMessage(ex);
-            LogServiceImpl::Log(level, full_message);
+            switch (level)
```

Comment:
And again...

Another possible DRY optimization: make all the overloaded calls to Log call this one implementation with whatever the default arguments are for the missing arguments.

Reply:
Thanks for the feedback. For the second suggestion, calling one Log implementation with default arguments is resulting in failure of existing LogService tests since the value in the 'message' comes out to be different when default input values come into picture. (For example, if we pass nullptr as default value in exception pointer, 'GetExceptionMessage()' function returns 'none' and as a result, this gets added in the message)